back/app/validators/multiloc_validator.rb

```diff
@@ -1,72 +1,54 @@
 class MultilocValidator < ActiveModel::EachValidator
-
-  def validate_each record, attribute, value
+  def validate_each(record, attribute, value)
     validate_presence record, attribute, value
     return if record.errors.present? || value.nil?
 
     validate_supported_locales record, attribute, value
     validate_values_not_all_strings record, attribute, value
-    validate_length record, attribute, value
+    validate_html record, attribute, value
   end
 
-  def validate_presence record, attribute, value
+  private
+
+  def validate_presence(record, attribute, value)
     if (options[:presence] && !value.kind_of?(Hash)) || (!options[:presence] && !(value.kind_of?(Hash) || value.nil?))
-      record.errors[attribute] << (options[:message] || "is not a translation hash")
+      record.errors[attribute] << (options[:message] || 'is not a translation hash')
       return
     end
 
     sanitizer = SanitizationService.new
-    if options[:presence] && value.values.all?{|text_or_html| !sanitizer.html_with_content?(text_or_html)}
-      record.errors.add(attribute, :blank, 
-        message: (options[:message] || "should be set for at least one locale"))
+    if options[:presence] && value.values.all? { |text_or_html| !sanitizer.html_with_content?(text_or_html) }
+      message = options[:message] || 'should be set for at least one locale'
+      record.errors.add attribute, :blank, message: message
     end
   end
 
-  def validate_values_not_all_strings record, attribute, value
-    if value&.values && !value.values.all?{|v| v.is_a? String}
-      locales = value.keys.select{|l| !value[l].is_a?(String)}
-      record.errors.add(attribute, :values_not_all_strings, message: "non-string values (for #{locales}) cannot be accepted. Either the key should be removed, or the value should be replaced by an empty string") 
+  def validate_values_not_all_strings(record, attribute, value)
```

Comment:
This name is confusing, no? I guess this is not the focus of this PR, but interested in what you thing. Does this in fact, validate `are_all_strings`?

Reply:
Yes correct, all values must be strings, you cannot have something like `{'en' => nil}` We used to have this in data and this had the tendency to cause all sorts of annoying bugs. Will change the name.

Reply:
One aspect of this method that @jinjagit and I discussed was the strange name. The method `validate_values_not_all_strings` does not validate that the values are "not all strings". It validates that all values "are strings", because it adds errors when not all values are strings. Therefore a better name would be `validate_values_are_strings`.